ui/components/multichain/network-list-menu/select-rpc-url-modal/select-rpc-url-modal.tsx

```diff
@@ -76,8 +94,29 @@ export const SelectRpcUrlModal = ({
               }),
             );
             dispatch(setActiveNetwork(rpcEndpoint.networkClientId));
-            dispatch(setEditedNetwork());
+            dispatch(
+              setEditedNetwork({ chainId: networkConfiguration.chainId }),
+            );
             dispatch(toggleNetworkMenu());
+            if (permittedAccountAddresses.length > 0) {
+              grantPermittedChain(
```

Comment:
why wouldn't we check if the chainId is already permitted before calling to grant it? I mean I'm sure its idempotent but seems like we should check first.